internal/cmd/controller/controllers/bundle/controller.go

```diff
@@ -342,6 +342,7 @@ func (h *handler) updateStatusAndTargets(status *fleet.BundleStatus, allTargets
 	}
 
 	for _, partition := range partitions {
+		partition := partition
```

Comment:
Is there some issue here, while partitions is not a pointer array, the next line `partition.Targets` is, but I can't spot a problem on first glance.

Reply:
To be honest, I do don't know for sure. But apparently due to the move to Go 1.21, or perhaps because I'm running a more recent version of `golangci-lint`, it became one for `golangci-lint`:

```go
internal/cmd/controller/controllers/bundle/controller.go:358:40: G601: Implicit memory aliasing in for loop. (gosec)
                        updateTarget(currentTarget, status, &partition.Status)
                                                            ^
internal/cmd/controller/controllers/bundle/controller.go:361:37: G601: Implicit memory aliasing in for loop. (gosec)
                if target.UpdateStatusUnavailable(&partition.Status, partition.Targets) {
```

Do you think `golangci-lint` is wrong about it?

Reply:
I researched it and yes, we should hold off on golangci-lint 1.54.2: `https://github.com/golangci/golangci-lint/issues/4037`
It's probably another false positive from Gosec.

Unless someone can show me invalid usage of the pointer, like adding it to an array `allthesameptrs = append(allthesameptrs, &partition)` or using it in a closure. In that case we would have a serious bug in our targeting and would probably need to backport fixes.


We can also discuss if we always want to copy pointers in a for loop to be defensive about it, but 🤷

Reply:
I'm happy to discuss, but it does not have to be in this PR. I've not found an issue that would require a fix and therefore removed the two changed from the PR.